blogs/views.py

```diff
@@ -1,3 +1,265 @@
+
+        Returns:
+            Response: The HTTP response object.
+        """
+
+        post = Posts.objects.filter(post_id=post_id, user=request.user).first()
+        if not post:
+            return cr.error(message="Post not found.")
+
+        post.delete()
+        return cr.success(message="Post deleted successfully.")
+
+
+class CreateCommentView(APIView):
+    permission_classes = [IsAuthenticated]
+
+    serializer_class = CommentSerializer
+
+    def post(self, request: Request) -> Response:
+        """
+        Post a new comment in a post.
+
+        Args:
+            request (Request): The HTTP request object.
+
+        Returns:
+                Response: The HTTP response object.
+        """
+
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        serializer.save(user=request.user)
+
+        return cr.success(
+            data=serializer.data, message="New comment added successfully!"
+        )
+
+    def delete(self, request: Request, id) -> Response:
+        """
+        Delete the comments.
+
+        Args:
+            request (Request): The HTTP request object.
+
+        Returns:
+            Response: The HTTP response object.
+        """
+
+        comment = Comments.objects.filter(id=id, user=request.user).first()
+        if not comment:
+            return cr.error(message="Comment not found.")
+
+        comment.delete()
+        return cr.success(message="Comment deleted successfully.")
+
+
+class EditCommentView(APIView):
+    permission_classes = [IsAuthenticated]
+
+    serializer_class = CommentEditSerializer
+
+    def put(self, request: Request, id) -> Response:
+        """
+        Update the comments.
+
+        Args:
+            request (Request): The HTTP request object.
+
+        Returns:
+            Response: The HTTP response object.
+        """
+
+        comment = Comments.objects.filter(id=id, user=request.user).first()
+        if not comment:
+            return cr.error(message="Comment not found.")
+
+        serializer = self.serializer_class(comment, data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        serializer.save()
+
+        return cr.success(data=serializer.data, message="Comment updated successfully!")
+
+
+class ReadPostView(APIView):
+    serializer_class = PostSerializer
+
+    def get(self, request: Request, post_id) -> Response:
+        """
+        Get the information about a blog post.
+
+        Args:
+            request (Request): The HTTP request object.
+
+        Returns:
+            Response: The HTTP response object.
+        """
+
+        posts = Posts.objects.filter(post_id=post_id).first()
+        if not posts:
+            return cr.error(message="Post not found.")
+        serializer = self.serializer_class(posts)
+        return cr.success(data=serializer.data, message="Blog fetched successfully!")
+
+
```

Comment:
This view is useless, we could have used the `GetBlogView` for this implementation as well!

Reply:
I thought we couldn't use two get functions in the same class. lol